ProcessLib/Process.cpp

```diff
@@ -169,23 +201,46 @@ void Process::constructDofTable()
     _mesh_subset_all_nodes =
         std::make_unique<MeshLib::MeshSubset>(_mesh, &_mesh.getNodes());
 
-    // Collect the mesh subsets in a vector.
+    // Vector of mesh subsets.
     std::vector<MeshLib::MeshSubsets> all_mesh_subsets;
-    for (ProcessVariable const& pv : _process_variables)
+
+    // Vector of the number of variable components
+    std::vector<int> vec_var_n_components;
+    if (_use_monolithic_scheme)
+    {
+        // Collect the mesh subsets in a vector.
+        for (ProcessVariable const& pv : _process_variables[0])
+        {
+            std::generate_n(
+                std::back_inserter(all_mesh_subsets),
+                pv.getNumberOfComponents(),
+                [&]() {
+                    return MeshLib::MeshSubsets{_mesh_subset_all_nodes.get()};
+                });
+        }
+
+        // Create a vector of the number of variable components
+        for (ProcessVariable const& pv : _process_variables[0])
+            vec_var_n_components.push_back(pv.getNumberOfComponents());
+    }
+    else  // for staggered scheme
     {
+        // Assuming that all equations of the coupled process use the same
+        // element order. Other cases can be considered by overloading this
+        // member function in the derived class.
+
+        // Collect the mesh subsets in a vector.
         std::generate_n(
             std::back_inserter(all_mesh_subsets),
-            pv.getNumberOfComponents(),
+            _process_variables[0][0].get().getNumberOfComponents(),
```

Comment:
get() is not needed.

Reply:
`_process_variables[0][0]` has the type of  `std::reference_wrapper`. Removing `get()``results in a compilation error. Do you have any other suggestion?

Reply:
Oh, my mistake. I thought it would be a `unique_ptr`. All fine.